### PR TITLE
Add make deps command

### PR DESCRIPTION
**Pull request by paulbellamy**
Works for me, but no idea if it is the correct thing to do for python...

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,6 +35,11 @@ def local_file(name):
     ],
     install_requires=[
     ],
+    extras_require={
+        'dev': [
+            'flake8',
+        ],
+    },
```
Comment (jml) [inline setup.py]: This particular change isn't strictly necessary for the Makefile stuff, but it's a bit more idiomatic. requirements.txt files are more often used for pinning versions in an application than for library dependencies.

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,4 @@ __pycache__
 build/
 dist/
 .uptodate
+/.env
```
Comment (paulbellamy) [inline .gitignore]: does `/.env` mean what you think?

File: circle.yml [hunk 3]
```diff
@@ -9,8 +9,7 @@ machine:
 
 dependencies:
   override:
-    - pip install -e .
-    - pip install -r dev-requirements.txt
+    - pip install -e .[dev]
```
Comment (paulbellamy) [inline circle.yml]: no `make deps` here for consistency?

Comment (jml) [inline circle.yml]: Good point! Done.

File: .gitignore (hunk 2 above)
Comment (jml) [inline .gitignore]: Yep.

**Pull request merged**